File: app/src/main/java/com/example/debtcalculator/data/Group.kt
package com.example.debtcalculator.data

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.annotations.JsonAdapter
import java.lang.reflect.Type

data class Group (
    var id: Int,
    var name: String,
    var users: MutableSet<User>,
    var expenses: MutableList<Expense>,
    var description: String,
    @JsonAdapter(IntToBooleanAdapter::class)
    var isClosed: Boolean,
    var messages: List<Message>,
    var image: String?
)

class IntToBooleanAdapter : JsonDeserializer<Boolean> {
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext
    ): Boolean {
        return json.asInt == 1
    }
}

File: backend/data/Group.js
class Group {
    constructor(name, users = [], expenses = [], messages = [], id) {
        this.name = name;
        this.users = users;
        this.expenses = expenses;
        this.messages = messages;
        this.id = id;
    }

    toJSON() {
        return {
            name: this.name,
            users: this.users.toJSON(),
            expenses: this.expenses.toJSON(),
            messages: this.messages,
            id: this.id
        };
    }
}

module.exports = Group;